.github/workflows/mypy.yml

```diff
@@ -0,0 +1,32 @@
+# Static type checking
+
+name: Static type checking
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  test:
+    name: Static type checking
+    runs-on: ubuntu-latest
+    container: python:3.13-alpine3.21
```

Comment:
Not sure if we want to use alpine3.21 here, renovate bots won't bump 3.21. Maybe just alpine or use `setup-python`?

Reply:
I'm happy to change it of course, just wanted to point out that sticking to a specific alpine version was not my own decision.

Reply:
Ok, it's fine then. We can tackle it separately.